api/envoy/extensions/filters/http/rbac/v3/rbac.proto

```diff
@@ -18,19 +22,41 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#extension: envoy.filters.http.rbac]
 
 // RBAC filter config.
+// [#next-free-field: 6]
 message RBAC {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.http.rbac.v2.RBAC";
 
   // Specify the RBAC rules to be applied globally.
   // If absent, no enforcing RBAC policy will be applied.
   // If present and empty, DENY.
-  config.rbac.v3.RBAC rules = 1;
+  config.rbac.v3.RBAC rules = 1
+      [(udpa.annotations.field_migrate).oneof_promotion = "rules_specifier"];
+
+  // The match tree to use when resolving RBAC action for incoming requests. Requests do not
+  // match any matcher will be denied.
+  // If absent, no enforcing RBAC matcher will be applied.
+  // If present and empty, deny all requests.
+  // If both rules and matcher are configured, the matcher will be ignored.
```

Comment:
Maybe we should do this the opposite way, where matcher takes precedence over old-style rules?  That will provide a safe way to transition to the new field: control planes can populate both fields, and old clients will continue to use the old rules field, but newer clients will use the new matcher field.

Reply:
The matching API is still unstable. In the previous work, we have added matcher for filter chain, and it is also a secondary option which is an inferior of filter chain match.

Reply:
I understand that it's not yet stable, but it's still up to the control plane as to whether or not it wants to use this unstable feature.  If it doesn't want to do that, it can just not populate this field.

Reply:
Ok, I will change the behavior.